pyigm/cgm/models.py

```diff
@@ -0,0 +1,240 @@
+
+from astropy import units as u
+from astropy import constants as const
+
+try:
+    basestring
+except NameError:  # For Python 3
+    basestring = str
+
+def rad3d2(xyz):
+    return xyz[0]**2 + xyz[1]**2 + xyz[-1]**2
+
+# Constants
+m_p = const.m_p.cgs.value # g
+
+class CGMModel(object):
+    """Model of the CGM
+    Wraps multiple phases together
+
+    Parameters:
+    -----------
+    """
+    def __init__(self, **kwargs):
+        # Groups
+        self._pdict = {}   # Dict for the phases
+
+    def __getitem__(self, key):
+        """ Access the DB groups
+
+        Parameters
+        ----------
+        key : str
+
+        Returns
+        -------
+
+        """
+        # Check
+        if not isinstance(key, basestring):
+            raise IOError("Item must be str")
+        # Try to access the dict
+        try:
+            return self._pdict[key]
+        except KeyError:
+            raise IOError("Input phase={:s} is not loaded in the model".format(key))
+
+    def __repr__(self):
+        txt = '<{:s}: '.format(self.__class__.__name__)
+        # Phases
+        txt += '   Phases modeled = {} \n'.format(self._pdict.keys())
+        txt += '>'
+        return (txt)
+
+class CGMPhase(object):
+    """ Model of a single phase of the CGM
+    Parameters:
+    -----------
+    phase : str
+      'hot'
+
+    """
+    def __init__(self, phase, **kwargs):
+        # Check
+        assert phase in ['hot']
+        #
+        self.phase = phase
+        self.mass = 0.   # Total mass in Solar masses
+        self.r_cgm = 0.  # Radius of the CGM
+
+    def nH(self, xyz):
+        """
+        Parameters
+        ----------
+        xyz : ndarray
+          Coordinate(s) in kpc
+
+        Returns
+        -------
+        nH : float or ndarray
+          Hydrogen number density in cm**-3
+        """
+        pass
+
+
+class ModifiedNFW(CGMPhase):
+    """ Generate a modified NFW model, e.g. Mathews & Prochaska 2017
+    for the hot gas.  Currently only valid for z=0
+
+    Parameters:
+    -----------
+    M_halo : float, optional
+      log10 of the Halo mass (solar masses)
+    c : float, optional
+      concentration of the halo
+    f_hot : float, optional
+      Fraction of the brayons in this hot phase
+    alpha : float, optional
+      Parameter to modify NFW profile power-law
+    y0 : float, optional
+      Parameter to modify NFW profile position
+    """
+    def __init__(self, M_halo=12.2, c=7.67, f_hot=0.9, alpha=0., y0=1., **kwargs):
+        # Init
+        CGMPhase.__init__(self, 'hot')
+        # Param
+        self.M_halo = M_halo
+        self.c = c
+        self.alpha = alpha
+        self.y0 = y0
+        self.f_hot = f_hot
+        # Init more
+        self.setup_param()
+
+    def setup_param(self):
+        """ Setup key parameters of the model
+        """
+        # Cosmology
+        self.H0 = 70. *u.km/u.s/ u.Mpc
+        self.fb = 0.16       # Baryon fraction
+        self.rhoc = 9.2e-30 * u.g / u.cm**3
```

Comment:
I believe references for these numbers are in Mathews & Prochaska (2017), but maybe some short comments here listing where they come from so that people could easily figure out the related papers in the future?